consensus/src/threshold_simplex/actors/voter/actor.rs

```diff
@@ -1057,50 +1073,59 @@ impl<
             return;
         };
 
-        // Verify signature
+        // Parse partial signature to get index
         let Some(signature) = Eval::deserialize(&nullify.view_signature) else {
-            debug!(
-                public_key_index,
-                "partial signature is not formatted correctly"
-            );
             return;
         };
         if signature.index != public_key_index {
-            debug!(
-                public_key_index,
-                partial_signature = signature.index,
-                "invalid signature index for nullify"
-            );
-            return;
-        }
-        let nullify_message = nullify_message(nullify.view);
-        if ops::partial_verify_message(
-            identity,
-            Some(&self.nullify_namespace),
-            &nullify_message,
-            &signature,
-        )
-        .is_err()
-        {
             return;
         }
+        let public = identity.evaluate(signature.index);
 
-        // Verify seed
-        let Some(seed) = Eval::deserialize(&nullify.seed_signature) else {
-            return;
-        };
-        if seed.index != public_key_index {
-            return;
-        }
-        let seed_message = seed_message(nullify.view);
-        if ops::partial_verify_message(identity, Some(&self.seed_namespace), &seed_message, &seed)
-            .is_err()
-        {
-            return;
-        }
+        // Verify signature
+        self.context.with_label("nullify").spawn({
```

Comment:
While I originally thought we should do this as a "stream wrapper", I found that the optimal approach was deep within consensus (after we filtered things we would otherwise never want to verify).

Reply:
This change isn't without tradeoffs, however. It is now possible that we may verify duplicate messages from the same user (whereas we otherwise would have updated the map upon successful processing).

This should only happen if messages are queued very close together but the point still stands.